### PR TITLE
Fix(wren-ui): use postgres as database

### PR DESCRIPTION
**Pull request by onlyjackfrost**
Fix schema issue to use PG & SQLite as our database.

Set sqlite as our default database.
To use PG as database
```
export DB_TYPE=pg
```

### CODE REVIEW

File: wren-ui/migrations/20240524044348_update_project_table.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = async function (knex, Promise) {
+  await knex.schema.alterTable('project', (table) => {
+    table.text('init_sql').alter();
```
Comment (onlyjackfrost) [inline wren-ui/migrations/20240524044348_update_project_table.js:7]: alter column type from json to text, init_sql should be text

File: wren-ui/migrations/20240524071859_update_thread_table.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = async function (knex, promise) {
+  // drop foreign key constraint before altering column type to prevent data loss
```
Comment (onlyjackfrost) [inline wren-ui/migrations/20240524071859_update_thread_table.js:6]: we have on_delete cascade on this constraint

File: wren-ui/src/apollo/server/repositories/threadResponseRepository.ts [hunk 3]
```diff
@@ -105,11 +113,16 @@ export class ThreadResponseRepository
...
     const camelCaseData = mapKeys(data, (_value, key) => camelCase(key));
-    // JSON.parse detail and error
-    return {
-      ...camelCaseData,
-      detail: camelCaseData.detail ? JSON.parse(camelCaseData.detail) : null,
-      error: camelCaseData.error ? JSON.parse(camelCaseData.error) : null,
-    } as ThreadResponse;
+    const formattedData = mapValues(camelCaseData, (value, key) => {
+      if (['error', 'detail'].includes(key)) {
+        if (typeof value === 'string') {
```
Comment (onlyjackfrost) [inline wren-ui/src/apollo/server/repositories/threadResponseRepository.ts:118]: The value from Sqlite will be string type, while the value from PG is JSON object

File: wren-ui/migrations/20240524071859_update_thread_table.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  await knex.schema.alterTable('thread_response', (table) => {
+    table.dropForeign('thread_id');
+  });
+  await knex.schema.alterTable('thread', (table) => {
+    table.text('sql').alter();
```
Comment (wwwy3y3) [inline wren-ui/migrations/20240524071859_update_thread_table.js:11]: Comment: Change `sql` data type from `string` to `text`

File: wren-ui/src/apollo/server/repositories/projectRepository.ts [hunk 5]
```diff
@@ -63,8 +64,11 @@ export class ProjectRepository
     }
     const camelCaseData = mapKeys(data, (_value, key) => camelCase(key));
     const formattedData = mapValues(camelCaseData, (value, key) => {
-      if (['configurations', 'extensions'].includes(key)) {
-        return JSON.parse(value);
+      if (key === 'configurations') {
+        return isEmpty(value) ? {} : JSON.parse(value);
+      }
+      if (key === 'extensions') {
+        return isEmpty(value) ? [] : JSON.parse(value);
```
Comment (wwwy3y3) [inline wren-ui/src/apollo/server/repositories/projectRepository.ts:71]: @onlyjackfrost nit: maybe add comments to explain why doing `isEmpty` check

File: wren-ui/src/apollo/server/repositories/threadResponseRepository.ts [hunk 6]
```diff
@@ -70,10 +70,18 @@ export class ThreadResponseRepository
       })
       .map((res) => {
         // JSON.parse detail and error
+        const detail =
+          res.detail && typeof res.detail === 'string'
```
Comment (onlyjackfrost) [inline wren-ui/src/apollo/server/repositories/threadResponseRepository.ts:74]: The return data type from pg is object but the data type from sqlite is string.
We only need to parse the value that comes from SQLite.

File: wren-ui/src/apollo/server/repositories/threadResponseRepository.ts (hunk 3 above)
Comment (wwwy3y3) [reply wren-ui/src/apollo/server/repositories/threadResponseRepository.ts:118]: Maybe add a comment to this line?

**Review by wwwy3y3 — COMMENTED** [review]
Others LGTM

**Review by wwwy3y3 — APPROVED** [review]
LGTM

**Pull request merged**